src/components/AnalysisPage/UnitsSelector/UnitsSelector.jsx

```diff
@@ -8,6 +8,25 @@ const UnitsSelector = ({ partId }) => {
   const [units, setUnits] = useState(undefined);
   const [showSuccess, setShowSuccess] = useState(false);
 
+  useEffect(() => {
+    const getCurrentUnits = async () => {
+      if (partId) {
+        try {
+          const response = await fetch(`/api/v0/parts/${partId}`);
+          if (!response.ok) {
+            throw new Error("Failed to get part data.");
+          }
+          const data = await response.json();
+          setUnits(data.units);
+        } catch (error) {
+          console.error("Error getting part data:", error);
```

Comment:
This would maybe be good to have an error message popup but dont worry about this one right now because I think this call might get moved to a higher component as you start building out the analysis components